shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -77,6 +87,53 @@ typedef NS_ENUM(NSUInteger, FlutterTextAffinity) {
   return flutter::TextRange([base unsignedLongValue], [extent unsignedLongValue]);
 }
 
+// Returns the autofill hint content type, if specified; otherwise nil.
+static NSString* GetAutofillContentType(NSDictionary* autofill) {
+  NSArray<NSString*>* hints = autofill[kAutofillHints];
+  return hints.count > 0 ? hints[0] : nil;
+}
+
+// Returns YES if configuration describes a field for which autocomplete should be enabled for
+// the specified TextInputConfiguration. Autocomplete is enabled by default, but will be disabled
+// if the field is password-related, or if the configuration contains no autofill settings.
+static BOOL EnableAutocompleteForTextInputConfiguration(NSDictionary* configuration) {
+  // Disable if obscureText is set.
+  if ([configuration[kSecureTextEntry] boolValue]) {
+    return NO;
+  }
+
+  // Disable if autofill properties are not set.
+  NSDictionary* autofill = configuration[kAutofillProperties];
+  if (autofill == nil) {
+    return NO;
+  }
+
+  // Disable if autofill properties indicate a username/password.
+  NSString* contentType = GetAutofillContentType(autofill);
+  if ([contentType isEqualToString:@"password"] || [contentType isEqualToString:@"username"]) {
+    return NO;
+  }
+  return YES;
+}
+
+// Returns YES if configuration describes a field for which autocomplete should be enabled.
+// Autocomplete is enabled by default, but will be disabled if the field is password-related, or if
+// the configuration contains no autofill settings.
```

Comment:
nit: maybe link to the issue or provide a simple explanation as to why this is disabled for username/passwords?

Reply:
Good call; done. Also filed https://github.com/flutter/flutter/issues/120252, which should be relatively straightforward and nice to have.